cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/upgrade/DataMigration26.java

```diff
@@ -0,0 +1,282 @@
+      "-", ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE),
+
+      "gateway", ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE,
+                                 Constants.Metrics.Tag.COMPONENT, Constants.Gateway.METRICS_CONTEXT,
+                                 Constants.Metrics.Tag.HANDLER, Constants.Gateway.STREAM_HANDLER_NAME)
+    );
+
+  private Map<String, MetricsTable> scopeToAggregatesTable = Maps.newHashMap();
+  private Map<String, EntityTable> scopeToEntityTable = Maps.newHashMap();
+  private Map<String, MetricsEntityCodec> scopeToCodec = Maps.newHashMap();
+
+  public DataMigration26(final CConfiguration cConf, final DatasetFramework dsFramework,
+                         DefaultMetricDatasetFactory factory) {
+
+    //todo - change hard-coded table name to read from configuration instead.
+    LOG.info("Initializing Data Migration.");
+    this.dsFramework = dsFramework;
+    for (String scope : scopes) {
+      scopeToEntityTable.put(scope, new EntityTable(getOrCreateMetricsTable(scope + ENTITY_TABLE_NAME_SUFFIX,
+                                                                            DatasetProperties.EMPTY)));
+      scopeToAggregatesTable.put(scope, getOrCreateMetricsTable(scope + AGGREGATES_TABLE_NAME_SUFFIX,
+                                                                DatasetProperties.EMPTY));
+      scopeToCodec.put(scope, new MetricsEntityCodec(scopeToEntityTable.get(scope),
+                                                     UpgradeMetricsConstants.DEFAULT_CONTEXT_DEPTH,
+                                                     UpgradeMetricsConstants.DEFAULT_METRIC_DEPTH,
+                                                     UpgradeMetricsConstants.DEFAULT_TAG_DEPTH));
+    }
+    aggMetricStore = new DefaultMetricStore(factory, new int[]{Integer.MAX_VALUE});
+  }
+
+  public void decodeAggregatesTable26() {
+    try {
+      for (String scope : scopes) {
+        Scanner scanner = scopeToAggregatesTable.get(scope).scan(null, null, null, null);
+        MetricsEntityCodec codec = scopeToCodec.get(scope);
+        LOG.info("Decoding for scope " + scope + "on table " + scopeToAggregatesTable.get(scope).toString() +
+                   " using codec " + codec.toString());
+        Row row;
+        while ((row = scanner.next()) != null) {
+          byte[] rowKey = row.getRow();
+          int offset = 0;
+          String context = codec.decode26(MetricsEntityType.CONTEXT, rowKey, offset);
+          offset += codec.getEncodedSize26(MetricsEntityType.CONTEXT);
+          String metricName = codec.decode26(MetricsEntityType.METRIC, rowKey, offset);
+          offset += codec.getEncodedSize26(MetricsEntityType.METRIC);
+          String runId = codec.decode26(MetricsEntityType.RUN, rowKey, offset);
+          constructMetricValue(scope, context, metricName, runId, row.getColumns().entrySet().iterator());
```

Comment:
construct should not emit. It would be a lot more readable if we do:
``` java
public void migrateData() {
   for (scope in scopes) {
      migrateScope(scope);
   }
}

private void migrateScope(String scope) {
   Scanner scanner = ...
   while ((row = scanner.next()) != null) {
      MetricValue metricValue = parseMetricValue(row, ...);
      if (metricValue != null) {
         write(metricValue);
      }
   }
}
```

and so on. Please refactor to make it more readable